forecasting/src/autogluon/forecasting/predictor.py

```diff
@@ -0,0 +1,420 @@
+        """Returns the list of model names trained in this `predictor` object."""
+        return self._trainer.get_model_names()
+
+    def predict(
+        self,
+        data: TimeSeriesDataFrame,
+        model: Optional[str] = None,
+        **kwargs,
+    ) -> TimeSeriesDataFrame:
+        """Return quantile and mean forecasts given a dataset to condition on.
+
+        Parameters
+        ----------
+        data: TimeSeriesDataFrame
+            Time series data to forecast with.
+        model: str, default=None
+            Name of the model that you would like to use for forecasting. If None, it will by default use the
+            best model from trainer.
+        """
+        return self._learner.predict(data, model=model, **kwargs)
+
+    def evaluate(self, data: TimeSeriesDataFrame, **kwargs):
+        """Evaluate the performance for given dataset, computing the score determined by `self.eval_metric`
+        on the given data set, and with the same `prediction_length` used when training models."""
+        return self._learner.score(data, **kwargs)
+
+    def score(self, data: TimeSeriesDataFrame, **kwargs):
+        """See `evaluate`"""
+        return self.evaluate(data, **kwargs)
+
+    @classmethod
+    def load(cls, path: str) -> "ForecastingPredictor":
+        """Load an existing ForecastingPredictor from output_directory."""
+        if not path:
+            raise ValueError("`path` cannot be None or empty in load().")
+        path = setup_outputdir(path, warn_if_exist=False)
+
+        logger.info(f"Loading predictor from path {path}")
+        learner = AbstractLearner.load(path)
+        predictor = load_pkl.load(path=learner.path + cls.predictor_file_name)
+        predictor._learner = learner
+        return predictor
+
+    def save(self) -> None:
+        """Save this predictor to file in directory specified by this Predictor's `output_directory`.
+        Note that `fit()` already saves the predictor object automatically
+        (we do not recommend modifying the Predictor object yourself as it tracks many trained models).
+        """
+        tmp_learner = self._learner
+        self._learner = None
+        save_pkl.save(path=tmp_learner.path + self.predictor_file_name, object=self)
+        self._learner = tmp_learner
+
+    def info(self) -> Dict[str, Any]:
+        return self._learner.get_info(include_model_info=True)
+
+    def get_model_best(self) -> str:
+        """Returns the name of the best model from trainer"""
+        return self._trainer.get_model_best()
+
+    def leaderboard(self, data: Optional[TimeSeriesDataFrame] = None) -> pd.DataFrame:
+        """Return a leaderboard showing the performance of every trained model
+
+        Parameters
+        ----------
+        data: TimeSeriesDataFrame
+            dataset used for additional evaluation. If None, the validation set used during training will
+            be used.
+        """
+        return self._learner.leaderboard(data)
```

Comment:
Tabular describes each column present in the leaderboard. We may want to do the same here.